### ISSUE TITLE
In case you would like to add more

### ISSUE THREAD

**Issue by pachacamac** [dirkschumacher/genossenschaften#1]
check out https://www.wohnungsbaugenossenschaften.de/genossenschaften/alle-genossenschaften

**Comment by dirkschumacher** [comment]
Nice. Maybe a case for wikidata 🤔

**Comment by dirkschumacher** [comment]
@k-nut I have never imported data into wikidata, but do you think we could have the data there?

**Comment by k-nut** [comment]
Yes, that sounds good! Right now there are only 33 Housing Cooperatives in Wikidata at all as you can check [here](https://query.wikidata.org/#%23Cats%0ASELECT%20%3Fitem%20%3FitemLabel%20%0AWHERE%20%0A%7B%0A%20%20%3Fitem%20wdt%3AP31%20wd%3AQ562166.%0A%20%20SERVICE%20wikibase%3Alabel%20%7B%20bd%3AserviceParam%20wikibase%3Alanguage%20%22%5BAUTO_LANGUAGE%5D%2Cen%22.%20%7D%0A%7D).

It would probably be pretty easy to add the Genossenschaften from the website via quickstatements as described [here in the documentation](https://www.wikidata.org/wiki/Help:QuickStatements#Item_creation).

All of this is of course if we think that the license allows this. If there is a license on the data, I'm not sure if this collection of facts is copyrightable but INAL so ¯\_(ツ)_/¯

**Comment by pachacamac** [comment]
`[...document.querySelectorAll('.address')].map(e => e.innerText.replace(/\s+/g,' '))` should basically give you roughly what you need ... but I'm sure Wikidata has better tools at hand. I believe a list of company addresses ~~is not really~~ should really not be copyright-able? Otherwise could probably pull from yellow pages etc.

**Comment by dirkschumacher** [comment]
The information that is needed is the website, name, address and ideally if they still admit new members and maybe what the average net rent is. That would be enough to built a small tool.

**Comment by pachacamac** [comment]
website, name, address, email, phone number are all there and easy to extract. Average rent however is more difficult.

Here's a snippet to throw into the dev console to extract all that info as JSON:

    function directTextContent(element){
      return Array.prototype.filter.call(element.childNodes, function (element) {
        return element.nodeType === Node.TEXT_NODE;
      }).map(function (element) {
          return element.textContent;
      }).join("");
    }

    var objects = [...document.querySelectorAll('.address')].map(function(e){
      var address = directTextContent(document.querySelector('.address p')).trim().split(/\s{2,}/)
      var links = [...e.querySelectorAll('a')].map(a => a.href)
      var obj = {
        name: e.querySelector('strong').innerText,
        street: address[0],
        postalCode: address[1].match(/^(\d+) (.*)$/)[1],
        city: address[1].match(/^(\d+) (.*)$/)[2],
        telephone: address[2].substr(4),
        telefax: address[3].substr(4),
      }
      var email = links.find(a => a.match(/^mailto:/))
      if(email) obj.email = email.substr(7)
      var url = links.find(a => !a.match(/^mailto:/))
      if(url) obj.url = url
      return obj
    })

    console.log(JSON.stringify(objects))

**Comment by almereyda** [comment]
There is a good chance the people over at https://id22.net/, who are working on cohousing, can provide more pointers to also find self-organised housing projects that are instaurated as a legal cooperative, such as http://spreefeld-berlin.de/ or

- http://www.cohousing-berlin.de/de/projekte/andere-welt-strausberg
- http://www.cohousing-berlin.de/de/projekte/moeckernkiez
- http://www.cohousing-berlin.de/de/projekte/landhausprojekt-50km-vor-berlin

**Comment by k-nut** [comment]
There is also this website which lists quite a lot of initiatives: https://wohnungsbaugenossenschaften-berlin.info

**Comment by k-nut** [comment]
I have now imported all of the Genossenschaften using a slightly modified version of @pachacamac script from above into Wikidata. You can find the updated (global) list here: https://w.wiki/84z

**Comment by pachacamac** [comment]
Awesome @k-nut . Happy something was helpful :)

Btw are citylab-berlin.org and codefor.de/berlin somehow related?
